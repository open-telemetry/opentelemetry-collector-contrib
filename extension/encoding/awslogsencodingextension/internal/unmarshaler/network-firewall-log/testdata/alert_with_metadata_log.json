{
    "firewall_name": "test-firewall",
    "availability_zone": "us-east-1a",
    "event_timestamp": "2025-10-20T10:30:45.123Z",
    "event": {
        "event_type": "alert",
        "flow_id": 1234567890,
        "src_ip": "192.168.1.100",
        "src_port": 54321,
        "dest_ip": "10.0.1.50",
        "dest_port": 443,
        "proto": "TCP",
        "alert": {
            "action": "allowed",
            "signature": "ET MALWARE Suspicious TLS Connection",
            "signature_id": 2027758,
            "rev": 3,
            "gid": 1,
            "category": "Malware",
            "severity": 1,
            "metadata": {
                "affected_product": ["Windows", "Linux"],
                "attack_target": ["Server", "Workstation"],
                "deployment": ["Perimeter", "Internal"],
                "former_category": ["Trojan", "Backdoor"],
                "malware_family": ["Zeus", "CryptoLocker"],
                "performance_impact": ["High", "Medium"],
                "signature_severity": ["Critical", "High"],
                "created_at": ["2025-10-20T00:00:00Z", "2025-10-20T06:00:00Z"],
                "updated_at": ["2025-10-20T12:00:00Z", "2025-10-20T18:00:00Z"]
            }
        }
    }
}
