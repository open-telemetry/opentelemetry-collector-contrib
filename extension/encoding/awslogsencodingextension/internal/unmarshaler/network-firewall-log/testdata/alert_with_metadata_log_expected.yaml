resourceLogs:
  - resource:
      attributes:
        - key: cloud.provider
          value:
            stringValue: aws
        - key: aws.networkfirewall.name
          value:
            stringValue: test-firewall
        - key: cloud.availability_zone
          value:
            stringValue: us-east-1a
    scopeLogs:
      - scope:
          name: github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension
          version: ""
          attributes:
            - key: encoding.format
              value:
                stringValue: aws.networkfirewall
        logRecords:
          - observedTimeUnixNano: "0"
            timeUnixNano: "1760956245123000000"
            attributes:
              - key: aws.networkfirewall.event.type
                value:
                  stringValue: alert
              - key: aws.networkfirewall.flow_id
                value:
                  intValue: "1234567890"
              - key: source.address
                value:
                  stringValue: 192.168.1.100
              - key: source.port
                value:
                  intValue: "54321"
              - key: destination.address
                value:
                  stringValue: 10.0.1.50
              - key: destination.port
                value:
                  intValue: "443"
              - key: network.transport
                value:
                  stringValue: TCP
              - key: aws.networkfirewall.alert.action
                value:
                  stringValue: allowed
              - key: aws.networkfirewall.alert.signature
                value:
                  stringValue: ET MALWARE Suspicious TLS Connection
              - key: aws.networkfirewall.alert.signature_id
                value:
                  intValue: "2027758"
              - key: aws.networkfirewall.alert.rev
                value:
                  intValue: "3"
              - key: aws.networkfirewall.alert.category
                value:
                  stringValue: Malware
              - key: aws.networkfirewall.alert.severity
                value:
                  intValue: "1"
              - key: aws.networkfirewall.alert.gid
                value:
                  intValue: "1"
              - key: aws.networkfirewall.alert.metadata.affected_product
                value:
                  arrayValue:
                    values:
                      - stringValue: Windows
                      - stringValue: Linux
              - key: aws.networkfirewall.alert.metadata.attack_target
                value:
                  arrayValue:
                    values:
                      - stringValue: Server
                      - stringValue: Workstation
              - key: aws.networkfirewall.alert.metadata.deployment
                value:
                  arrayValue:
                    values:
                      - stringValue: Perimeter
                      - stringValue: Internal
              - key: aws.networkfirewall.alert.metadata.former_category
                value:
                  arrayValue:
                    values:
                      - stringValue: Trojan
                      - stringValue: Backdoor
              - key: aws.networkfirewall.alert.metadata.malware_family
                value:
                  arrayValue:
                    values:
                      - stringValue: Zeus
                      - stringValue: CryptoLocker
              - key: aws.networkfirewall.alert.metadata.performance_impact
                value:
                  arrayValue:
                    values:
                      - stringValue: High
                      - stringValue: Medium
              - key: aws.networkfirewall.alert.metadata.signature_severity
                value:
                  arrayValue:
                    values:
                      - stringValue: Critical
                      - stringValue: High
              - key: aws.networkfirewall.alert.metadata.created_at
                value:
                  arrayValue:
                    values:
                      - stringValue: "2025-10-20T00:00:00Z"
                      - stringValue: "2025-10-20T06:00:00Z"
              - key: aws.networkfirewall.alert.metadata.updated_at
                value:
                  arrayValue:
                    values:
                      - stringValue: "2025-10-20T12:00:00Z"
                      - stringValue: "2025-10-20T18:00:00Z"
